Sources/Recycling/GridViews/Ads/Cell/AdsGridPromotedViewCell.swift

```diff
@@ -0,0 +1,106 @@
+            bottomGradientImageView.leadingAnchor.constraint(equalTo: imageBackgroundView.leadingAnchor),
+            bottomGradientImageView.trailingAnchor.constraint(equalTo: imageBackgroundView.trailingAnchor),
+
+            subtitleLabel.topAnchor.constraint(equalTo: topAnchor, constant: AdsGridViewCell.subtitleTopMargin),
+            subtitleLabel.leadingAnchor.constraint(equalTo: leadingAnchor, constant: .mediumSpacing),
+            subtitleLabel.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -25),
+            subtitleLabel.heightAnchor.constraint(equalToConstant: AdsGridViewCell.subtitleHeight),
+
+            titleLabel.topAnchor.constraint(equalTo: subtitleLabel.bottomAnchor, constant: AdsGridViewCell.titleTopMargin),
+            titleLabel.leadingAnchor.constraint(equalTo: leadingAnchor, constant: .mediumSpacing),
+            titleLabel.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -25),
+
+            iconImageView.leadingAnchor.constraint(equalTo: imageDescriptionView.leadingAnchor),
+            iconImageView.heightAnchor.constraint(equalToConstant: AdsGridViewCell.iconSize),
+            iconImageView.widthAnchor.constraint(equalToConstant: AdsGridViewCell.iconSize),
+            iconImageView.centerYAnchor.constraint(equalTo: imageDescriptionView.centerYAnchor),
+
+            imageTextLabel.leadingAnchor.constraint(equalTo: iconImageView.trailingAnchor, constant: .mediumSpacing),
+            imageTextLabel.trailingAnchor.constraint(equalTo: imageDescriptionView.trailingAnchor),
+            imageTextLabel.centerYAnchor.constraint(equalTo: imageDescriptionView.centerYAnchor),
+
+            accessoryLabel.topAnchor.constraint(equalTo: imageTextLabel.bottomAnchor, constant: .mediumSpacing),
+            accessoryLabel.leadingAnchor.constraint(equalTo: imageDescriptionView.leadingAnchor),
+            accessoryLabel.trailingAnchor.constraint(equalTo: imageDescriptionView.trailingAnchor),
+            accessoryLabel.bottomAnchor.constraint(equalTo: imageDescriptionView.bottomAnchor),
+
+            imageDescriptionView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: .mediumSpacing),
+            imageDescriptionView.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -.mediumSpacing),
+            imageDescriptionView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -.mediumSpacing),
+
+            favoriteButton.topAnchor.constraint(equalTo: topAnchor, constant: .smallSpacing),
+            favoriteButton.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -.smallSpacing),
+            favoriteButton.widthAnchor.constraint(equalToConstant: 34),
+            favoriteButton.heightAnchor.constraint(equalTo: favoriteButton.heightAnchor)
+            ])
+    }
+}
+
+extension UIImage {
+    class func imageWithLayer(layer: CALayer) -> UIImage {
+        UIGraphicsBeginImageContextWithOptions(layer.bounds.size, layer.isOpaque, UIScreen.main.scale)
```

Comment:
I'm not sure if I'm correct, but wouldn't the gradient be pixelated/blurry when the cell is reused and the other image is smaller/bigger than the first one? Because it's an image and the size is set based on the layer's size

Reply:
If I'm not mistaken when you stretch a gradient it works fine since it fades as you stretch it. Since gradients are blurry by default. Maybe someone else knows more about it.

Reply:
You're probably right. Forgot that the gradient is transparent as well, which helps.